### PR TITLE
Option to enable specific language or ecosystem cataloger

### LINKED ISSUE #840: Enable/disable SBOM generation for specific language types
**What would you like to be added**:

The syft command line tool generates SBOM's for all types of languages - JS, Ruby, Python, Php, Golang, Java, Rust & base operating system packages. 

However, it would be nice to provide some command line options to syft, so that we can enable/disable the SBOM generation for specific language types

**Why is this needed**:

We at Deepfence (www.deepfence.io) are big fans of syft & gripe, and we would like to integrate them into our open source offering, ThreatMapper (https://github.com/deepfence/ThreatMapper).

We get requests from community that they know the language types of their images, and would like to run vulnerability scans for those language types only.

If we provide such an option, it would improve the speed of the vulnerability scans.

**Additional context**:

We are glad to provide a pull request for this feature.

### PR DESCRIPTION
**Pull request by ramanan-ravi**
Fixes #840 

If user knows the languages / ecosystem in the image or file system being scanned, user can run syft to generate sbom only for those specific languages / ecosystem.

Example command to enable a language or ecosystem cataloger
```
syft packages alpine:latest --enable-cataloger apkdb-cataloger
```

### CODE REVIEW

**Comment by kzantow** [comment]
This is looking pretty good, at first glance. I'll review more thoroughly later today. Note you'll need to sign-off and sign the commits: https://github.com/anchore/syft/blob/main/CONTRIBUTING.md#contributing-to-syft

File: README.md [hunk 1]
```diff
@@ -310,6 +310,24 @@ package:
     # same as -s ; SYFT_PACKAGE_CATALOGER_SCOPE env var
     scope: "squashed"
 
+  # enable specific language or ecosystem cataloger
+  # default: all catalogers are enabled by default
```
Comment (wagoodman) [inline README.md:314]: nit: mentioning that all catalogers are enabled by default is a little misleading. There is still a selection of which catalogers to use based off of the source (dir scan or image scan)

File: cmd/packages.go [hunk 2]
```diff
@@ -134,6 +134,11 @@ func setPackageFlags(flags *pflag.FlagSet) {
 		"exclude paths from being scanned using a glob expression",
 	)
 
+	flags.StringArrayP(
+		"enable-cataloger", "", nil,
```
Comment (wagoodman) [inline cmd/packages.go:138]: this could probably just be `catalogers` on the cli.

File: syft/pkg/cataloger/cataloger.go [hunk 3]
```diff
@@ -80,5 +80,28 @@ func AllCatalogers(cfg Config) []Cataloger {
...
+	}, cfg.EnabledCatalogers)
+}
+
+func filterCatalogers(catalogers []Cataloger, enabledCatalogers []string) []Cataloger {
+	// if enable-cataloger is not set, all applicable catalogers are enabled by default
+	if len(enabledCatalogers) == 0 {
+		return catalogers
+	}
+	var filteredCatalogers []Cataloger
+	for _, cataloger := range catalogers {
```
Comment (wagoodman) [inline syft/pkg/cataloger/cataloger.go:92]: we could add additional logic such that the string `-cataloger` is optional on each user-specified name, to be nicer to the user.

**Comment by wagoodman** [comment]
Open question: in the near future we will be deprecating the `packages` subcommand and instead suggesting for folks to use the root command. Why is this important? With this switch is also signaling the ability to leverage non-package catalogers, found here: https://github.com/anchore/syft/tree/main/syft/file .

I feel that this work should also consider how that should work (either in this PR or in a future state)

File: cmd/packages.go (hunk 2 above)
Comment (kzantow) [reply cmd/packages.go:138]: I was thinking the same thing here 👍

**Comment by houdini91** [comment]
Hi guys.
I was trying to understand the difference between the groups of catalogers that are assigned by default to images vs directories
I am not sure i understand the difference between the two defined groups ("installation" vs "index") , can anyone meaning this a bit for me?

* Should these group be configurable as well as apposed as hard coding the cataloger group by scheme? 
Of curse this suggestion is in some sense redundant because this PR will allow a full cataloger specific customization as well.

**Comment by houdini91** [comment]
Hi people.
Added a commit to fix the cataloger naming comment,
I pushed a PR on the deepfence:main from my fork (should i pushed somewhere else?).

[https://github.com/deepfence/syft/pull/1](PR)

**Comment by houdini91** [comment]
Update:
I opened a new PR - https://github.com/anchore/syft/pull/888/files
I apologize, but i wanted to rebase against the main and add a small `--cataloger-group` feature suggestion.

**Comment by wagoodman** [comment]
Hey! 👋  I tried to push a few updates to the PR branch without success:
```
$ git push origin HEAD:refs/pull/843/head

Enumerating objects: 238, done.
Counting objects: 100% (165/165), done.
Delta compression using up to 12 threads
Compressing objects: 100% (93/93), done.
Writing objects: 100% (95/95), 16.42 KiB | 840.00 KiB/s, done.
Total 95 (delta 73), reused 0 (delta 0), pack-reused 0
remote: Resolving deltas: 100% (73/73), completed with 36 local objects.
To github.com:anchore/syft.git
 ! [remote rejected]   HEAD -> refs/pull/843/head (deny updating a hidden ref)
error: failed to push some refs to 'github.com:anchore/syft.git'
```

Same story with a force push... I'm going to get this pulled into a in-repo branch to that I can apply some updates. I'll update with links here shortly.

**Pull request merged**